src/gluonts/dataset/repository/_uber.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+import json
+import tempfile
+import zipfile
+from pathlib import Path
+from urllib import request
+
+import pandas as pd
+
+from gluonts.dataset.repository._util import metadata
+
+
+def generate_uber_dataset(
+    dataset_path: Path, uber_freq: str, prediction_length: int
+):
+    subsets = {"daily": "1D", "hourly": "1H"}
+    assert (
+        uber_freq.lower() in subsets
+    ), f"invalid uber_freq='{uber_freq}'. Allowed values: {subsets.keys()}"
+    freq_setting = subsets[uber_freq.lower()]
+
+    # download the dataset and read the data
+    with tempfile.TemporaryDirectory() as dir_path:
+        temp_dir_path = Path(dir_path)
+        temp_zip_path = temp_dir_path / "uber-dataset.zip"
+        uber_url_path = (
+            "https://raw.githubusercontent.com/fivethirtyeight/"
+            "uber-tlc-foil-response/master/uber-trip-data/"
+            "uber-raw-data-janjune-15.csv.zip"
+        )
+        request.urlretrieve(uber_url_path, temp_zip_path)
+        with zipfile.ZipFile(temp_zip_path) as zf:
+            zf.extractall(path=temp_dir_path)
+        uber_file_path = (
+            temp_dir_path / "uber-dataset" / "uber-raw-data-janjune-15.csv"
+        )
```

Comment:
```suggestion
        uber_file_path = temp_dir_path / "uber-raw-data-janjune-15.csv"
```

I think the `uber-dataset` part needs to be removed.